interfaces/builtin/desktop_legacy.go

```diff
@@ -261,13 +269,28 @@ accept
 accept4
 `
 
+type desktopLegacyInterface struct {
+	commonInterface
+}
+
+func (iface *desktopLegacyInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	snippet := ""
+	for _, r := range getDesktopFileRules(plug.Snap().InstanceName()) {
+		snippet += r + "\n"
+	}
```

Comment:
It looks like you could replace this loop with `strings.Join(getDesktopFileRules(...), "\n")`, which will avoid a bunch of string copies.  That'd lose the final new line, but that doesn't look like a problem here.